flocker/control/_persistence.py

```diff
@@ -279,10 +279,13 @@ def update_leases(transform, persistence_service):
         persistence service has saved.
     """
     config = persistence_service.get()
-    new_config = config.set("leases", transform(config.leases))
-    d = persistence_service.save(new_config)
-    d.addCallback(lambda _: new_config.leases)
-    return d
+    new_leases = transform(config.leases)
+    if config.leases != new_leases:
```

Comment:
This appears to compare the whole dictionary, including all key and value pairs.  However, the `Leases.expire` code returns the `config.leases` object if no changes are made.

Hence, it should be faster to use `is not` here.